docs/plugins/criteria_for_core_components.md

```diff
@@ -0,0 +1,46 @@
+# Establishing Criteria for Core vs Non-Core Components
+
+## Overview
+As OpenSearch Dashboards continues to grow, we need clear criteria to determine what should be included in core versus non-core components. This decision framework will help maintain a focused, maintainable core while allowing for robust plugin ecosystem development.
```

Comment:
to clarify, "core" means the component or plugin exists in the opensearch-project/OpenSearch-Dashboards repo, and not necessarily in the `src/core` component?

Reply:
Right, core means its location inside OSD repo.